### PR TITLE
Fix bug that doesn't check for expired token

### PR DESCRIPTION
**Pull request by erdostom**

### CODE REVIEW

**Comment by erdostom** [comment]
Issue #9

**Comment by thenonameguy** [comment]
:+1:

File: lib/mkto_rest.rb [hunk 1]
```diff
@@ -20,7 +20,11 @@ def debug=(bool)
     end
 
     def authenticated?
-      return ! @token.empty?
+      return !@token.empty? && token_valid?
+    end
+
+    def token_valid?
+      return Time.now < @valid_until
```
Comment (thenonameguy) [inline lib/mkto_rest.rb]: There are two issues with this method:
- there is no need to explicitly write the `return` statement.
- `@valid_until` is nil until `authenticate` is called, so either you check for truthyness first, or provide a default Time value in the constructor.

My proposed implementation of `token_valid?` is:
```ruby
@valid_until && Time.now < @valid_until
```

**Comment by erdostom** [comment]
@thenonameguy Thank you, I incorporated your changes. koszi

**Comment by jalemieux** [comment]
Tests are falling.

**Comment by jalemieux** [comment]
Added test support, pull request merged in.

**Pull request merged**